src/backend/identities.rs

```diff
@@ -1296,6 +1479,104 @@ impl AppState {
+                64 => {
+                    // hex
+                    let bytes = match hex::decode(private_key) {
+                        Ok(bytes) => bytes,
+                        Err(_) => {
+                            return Err(WalletError::Custom("Failed to decode hex".to_string()))
+                        }
+                    };
+                    match PrivateKey::from_slice(bytes.as_slice(), Network::Testnet) {
+                        Ok(key) => Ok(key),
+                        Err(_) => {
+                            return Err(WalletError::Custom("Expected private key".to_string()))
+                        }
+                    }
+                }
+                51 | 52 => {
+                    // wif
+                    match PrivateKey::from_wif(private_key) {
+                        Ok(key) => Ok(key),
+                        Err(_) => return Err(WalletError::Custom("Expected WIF key".to_string())),
+                    }
+                }
+                _ => {
+                    return Err(WalletError::Custom(
+                        "Private key can't be decoded from hex or wif".to_string(),
+                    ));
+                }
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+        Self::add_identity_with_private_keys(&self, identity_id, private_keys, sdk).await
+    }
+
+    pub async fn add_identity_with_private_keys<'s>(
+        &self,
+        identity_id: &Identifier,
+        private_keys: Vec<PrivateKey>,
+        sdk: &Sdk,
+    ) -> Result<(), WalletError> {
+        let identity_fetch_result = Identity::fetch(sdk, *identity_id).await;
+        let identity = match identity_fetch_result {
+            Ok(Some(identity)) => identity,
+            Ok(None) => {
+                return Err(WalletError::Custom(
+                    "No identity found with that ID".to_string(),
+                ));
+            }
+            Err(e) => {
+                return Err(WalletError::Custom(format!(
+                    "Error fetching identity: {}",
+                    e
+                )));
+            }
+        };
+
+        let mut loaded_identity_lock = self.loaded_identity.lock().await;
+        *loaded_identity_lock = Some(identity.clone());
+
+        let mut identity_private_keys_map_lock = self.known_identities_private_keys.lock().await;
+
+        for private_key in private_keys.iter() {
+            let secp = Secp256k1::new();
+            let public_key_hash = private_key.public_key(&secp).pubkey_hash();
+            let key_id = identity
+                .public_keys()
+                .iter()
+                .filter_map(|(key_id, identity_public_key)| {
+                    if identity_public_key.public_key_hash().unwrap()
+                        == public_key_hash.to_byte_array()
+                    {
+                        Some(key_id)
+                    } else {
+                        None
+                    }
```

Comment:
```suggestion
                    (identity_public_key.public_key_hash().unwrap()  == public_key_hash.to_byte_array()).then_some(key_id)
```

Reply:
and I see an `unwrap` here: if you sure it's safe then please change to `expect` with a string why it shouldn't fail

Reply:
Cool